src/nvim/misc1.c

```diff
@@ -798,6 +798,9 @@ int prompt_for_number(int *mouse_used)
     cmdline_row = save_cmdline_row;
   }
   State = save_State;
+
+  ui_call_msg_clear();
```

Comment:
too early, doing something equivalent to `echo inputlist(["aa","bb"]) + inputlist(["cc", "dd"])` should not hide the message after first input. If the issue is specific to spell better fix it in `spell_suggest()`

Reply:
As far as I could tell, `echo inputlist(["aa", "bb"]) + inputlist(["cc", "dd"])` had the same behavior with and without my patch.

I've changed it and forced-pushed your suggestion anyway :).